plugins/dell-kestrel/fu-dell-kestrel-rmm.c

```diff
@@ -55,7 +54,7 @@ fu_dell_kestrel_rmm_write(FuDevice *device,
 			  FwupdInstallFlags flags,
 			  GError **error)
 {
-	return fu_dell_kestrel_ec_write_firmware_helper(device,
+	return fu_dell_kestrel_ec_write_firmware_helper(FU_DELL_KESTREL_EC(device),
```

Comment:
I've just realized that why I didn't explict the type for this helper because the rmm uses it too but rmm has its own usb device instead of using the proxy.

Reply:
will make a patch to bring it back.

Reply:
Hang on; that sounds wrong -- can you elaborate a bit pls

Reply:
> Hang on; that sounds wrong -- can you elaborate a bit pls

rmm has its own usbhid device to communicate with, while the update mechanism should be same as the ec flow.

Reply:
If we're not passing a FuDellKestrelEc then the method shouldn't be called `fu_dell_kestrel_ec_write_firmware_helper` -- and shouldn't be in `fu-dell-kestrel-ec.c` either... Would an appropriate fix be to create a FuDellKestrelDevice, and then have FuDellKestrelEc and FuDellKestrelRmm inherit it? Then instead of `fu_dell_kestrel_ec_hid_i2c_write` you'd have `fu_dell_kestrel_device_i2c_write` in the baseclass.

Reply:
Maybe the shared root device should be called FuDellKestrelEcHid or FuDellKestrelUsbHid? That kinda explains what the purpose of fu-dell-kestrel-ec-hid.h was...

Reply:
> If all that sounds scary I can help a bit.

Yeah that'll be great!

Reply:
> Maybe the shared root device should be called FuDellKestrelEcHid or FuDellKestrelUsbHid? That kinda explains what the purpose of fu-dell-kestrel-ec-hid.h was...

In general it abstracts the hid layer and the consumer could be `ec` as well as the `rmm` to leverage the same hid usage.

Reply:
https://github.com/fwupd/fwupd/pull/8170